stan/math/prim/meta/require_generics.hpp

```diff
@@ -935,6 +994,30 @@ template <template <class...> class TypeCheck, class... Check>
 using require_all_not_eigen_st
     = require_all_not_t<is_eigen_scalar_check<TypeCheck, Check>...>;
 
+template <template <class...> class TypeCheck, class... Check>
```

Comment:
What does the `s` mean?  I see `is_eigen_matrix_scalar_check` but I don't know what that means, either as it seems to be mixing types (matrix and scalar).

Reply:
The `st` stands for `scalar_type` it's doc'd on the website

http://mc-stan.org/math/d1/db9/group__require__meta.html

This PR is already pretty big so I'm going to open up another PR to move these